src/test/java/hbz/limetrans/filter/LibraryMetadataFilterTest.java

```diff
@@ -0,0 +1,81 @@
+package hbz.limetrans.filter;
+
+import hbz.limetrans.util.Helpers;
+
+import org.xbib.common.settings.Settings;
+
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+
+public class LibraryMetadataFilterTest {
+
+    private static final String BASE_PATH = "src/test/resources/filter";
+    private static final String INPUT_PATH = BASE_PATH + "/input/test.xml";
+    private static final String OUTPUT_PATH = BASE_PATH + "/output/test.json";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testMissingFile() throws IOException {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Could not process limetrans filter: no input specified.");
+
+        getFilter("missing-file");
+    }
+
+    @Test
+    public void testMissingFilter() throws IOException {
+        testEqualsReference("missing-filter");
+    }
+
+    @Test
+    public void testRecordId() throws IOException {
+        testEqualsReference("record-id", "001=ocm44959477");
+    }
+
+    @Test
+    public void testFieldEquals() throws IOException {
+        testEqualsReference("field-equals", "650??.a=Mathematics");
+    }
+
+    @Test
+    public void testFieldRegexp() throws IOException {
+        testEqualsReference("field-regexp", "856??.u=~book");
+    }
+
+    @Test
+    public void testAnyEquals() throws IOException {
+        testEqualsReference("any-equals", "Print version:");
+    }
+
+    @Ignore // org.culturegraph.mf.metamorph.functions.Regexp can't handle null values
+    @Test
+    public void testAnyRegexp() throws IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1607)